### PR TITLE
Enable exception logging in test fixture

### LINKED ISSUE #5758: swarm of fails in datalad/git-annex master branch testing
see https://github.com/datalad/git-annex/runs/2881114114?check_suite_focus=true  not yet sure what/why since we were green for https://github.com/datalad/datalad/pull/5691 PR

<details>
<summary>partial extract</summary> 

```shell
======================================================================
FAIL: datalad.support.tests.test_external_versions.test_external_versions_rogue_module
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/support/tests/test_external_versions.py", line 141, in test_external_versions_rogue_module
    assert_in('pickaboo', cml.out)
Warning: Error: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
-------------------- >> begin captured logging << --------------------
datalad: WARNING: Failed to import module verycustomrogue__ due to 
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: datalad.support.tests.test_github_.test__make_github_repos
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/support/tests/test_github_.py", line 113, in test__make_github_repos
    assert_in('Failed to create repository while using token 1to...: very bad status', cml.out)
Warning: Error: 'Failed to create repository while using token 1to...: very bad status' not found in '[WARNING] Failed to create repository while using token 1to...: \n'
-------------------- >> begin captured logging << --------------------
datalad.support.github_: WARNING: Failed to create repository while using token 1to...: 
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: datalad.support.tests.test_sshconnector.test_ssh_manager_close_no_throw
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/support/tests/test_sshconnector.py", line 202, in test_ssh_manager_close_no_throw
    assert_in('Failed to close a connection: oh I am so bad', cml.out)
AssertionError: 'Failed to close a connection: oh I am so bad' not found in '[DEBUG] Closing 1 SSH connections...\n[DEBUG] Failed to close a connection: \n'
-------------------- >> begin captured logging << --------------------
datalad.support.sshconnector: DEBUG: Closing 1 SSH connections...
datalad.support.sshconnector: DEBUG: Failed to close a connection: 
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: datalad.tests.test_utils.test_import_modules_fail
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/test_utils.py", line 1098, in test_import_modules_fail
    ok_startswith(failures[0], "Fail datalad.bogus: No module")
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 435, in ok_startswith
    msg="String %r doesn't start with %r" % (s, prefix))
AssertionError: String 'Fail datalad.bogus: ' doesn't start with 'Fail datalad.bogus: No module'

```
</details>

full logs could be found on smaug:/mnt/datasets/datalad/ci/git-annex/builds/2021/06/cron-20210622/build-ubuntu.yaml-317-a1f634dc-failed and I also see lots of hits for the 'pickaboo' in the logs for #5691:

```shell
(tinuous-dev) datalad@smaug:/mnt/datasets/datalad/ci/logs/2021/06/08/pr/5691$ git grep 'pickaboo'
3dcf2d3/github-Test on macOS-1425-failed/1_test (brew).txt:2021-06-08T12:51:55.5946970Z     assert_in('pickaboo', cml.out)
3dcf2d3/github-Test on macOS-1425-failed/1_test (brew).txt:2021-06-08T12:51:55.5948100Z AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
3dcf2d3/github-Test on macOS-1425-failed/2_test (snapshot).txt:2021-06-08T13:01:34.0297150Z     assert_in('pickaboo', cml.out)
3dcf2d3/github-Test on macOS-1425-failed/2_test (snapshot).txt:2021-06-08T13:01:34.0298580Z AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
3dcf2d3/github-Test on macOS-1425-failed/test (brew)/8_Run tests.txt:2021-06-08T12:51:55.5946960Z     assert_in('pickaboo', cml.out)
3dcf2d3/github-Test on macOS-1425-failed/test (brew)/8_Run tests.txt:2021-06-08T12:51:55.5948090Z AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
3dcf2d3/github-Test on macOS-1425-failed/test (snapshot)/8_Run tests.txt:2021-06-08T13:01:34.0297130Z     assert_in('pickaboo', cml.out)
3dcf2d3/github-Test on macOS-1425-failed/test (snapshot)/8_Run tests.txt:2021-06-08T13:01:34.0298570Z AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/57k0ylu8v3p39hqb.txt:[00:37:34]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/57k0ylu8v3p39hqb.txt:[00:37:34] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/fvlsmm3k5y2poi8v.txt:[00:44:40]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/fvlsmm3k5y2poi8v.txt:[00:44:40] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/jnnx886qnxc0j47q.txt:[00:44:50]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/jnnx886qnxc0j47q.txt:[00:44:50] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/n409j3f11lr50im0.txt:[00:32:40]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/n409j3f11lr50im0.txt:[00:32:40] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/p28vcoblx377qwwi.txt:[00:43:45]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/p28vcoblx377qwwi.txt:[00:43:45] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/1.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/1.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/2.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/2.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/3.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/3.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/6.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/6.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/7.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/7.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/8.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/8.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
```
so I guess it was not that clean

### PR DESCRIPTION
**Pull request by bpoldrack**
Exception logging is now disabled by default in datalad, but tests are
still making detailed assertions about what is logged how. Running them
currently requires that config to be set.

However, eventually tests need to be adjusted and enable the config
themselves, if they are making assertions about logged exceptions.

Closes #5758

### CODE REVIEW

File: datalad/__init__.py [hunk 1]
```diff
@@ -149,6 +149,8 @@ def set_envvar(v, val):
 [user]
 	name = DataLad Tester
 	email = test@example.com
+[datalad "log"]
+	exc = 1
```
Comment (yarikoptic) [inline datalad/__init__.py:153]: should the appveyor etc be cleaned up from that setting? (why to duplicate without need?)  
Could you please also add setting it within above `if 'GIT_HOME' in os.environ:` code block so no patching for that scenario would be needed?

Comment (mih) [reply datalad/__init__.py:153]: +1 from me on avoiding double-specification of configuration. This is always painful when an explicit switch is flipped and there is an identicial other setting that also needs to be altered.

**Comment by mih** [comment]
Test failures are both #5794

File: datalad/__init__.py (hunk 1 above)
Comment (yarikoptic) [reply datalad/__init__.py:155]: I have pushed tune ups for that + another tiny import issue detected while testing with `GIT_HOME` set.  Was testing only  on a single test: `GIT_HOME=$PWD/build/GIT_HOME python -m nose -s -v datalad/tests/test_utils.py:test_import_modules_fail`

**Comment by yarikoptic** [comment]
ok, everything is green besides known
```
FAIL: datalad.core.distributed.tests.test_clone.test_reckless(True,)
```
on OSX git hub actions (installs from dmg) with cmd:annex=8.20210631-g9905ec19a ... and passed on appveyor OSX with cmd:annex: 8.20210714 -- apparently there was a release I had missed (adjusted my "ping" email to Joey ) ... 

I will proceed with this

**Pull request merged**